modules/SofaGraphComponent/src/SofaGraphComponent/APIVersion.cpp

```diff
@@ -56,10 +57,13 @@ void APIVersion::checkInputData()
     if( !d_level.isSet() && name.isSet() ){
         d_level.setValue(getName());
     }
-    std::vector<std::string> allowedVersion = { "17.06", "17.12", "18.06", "18.12", SOFA_VERSION_STR } ;
+    std::vector<std::string> allowedVersion = { "17.06", "17.12", "18.06", "18.12", "19.06", "19.12", "20.06", "20.12", SOFA_VERSION_STR } ;
     if( std::find( allowedVersion.begin(), allowedVersion.end(), d_level.getValue()) == allowedVersion.end() )
     {
-        msg_warning() << "The provided level '"<< d_level.getValue() <<"' is now valid. " ;
+        auto allowedVersionStr = std::accumulate(std::next(allowedVersion.begin()), allowedVersion.end(), *(allowedVersion.begin()), [](const std::string & s, const std::string & v) {
+            return s + ", " + v;
+        });
+        msg_warning() << "The provided level '"<< d_level.getValue() <<"' is not valid. Allowed versions are [" << allowedVersionStr << "]." ;
```

Comment:
What happens next if the version is invalid? Wouldn't it be better to set a default value (e.g. `SOFA_VERSION_STR`)?

Reply:
Nothing will happen. The purpose of this component is not super clear, but if I understood correctly (maybe @hugtalbot or @guparan could correct me if I'm wrong):
 - Adding `<APIVersion level="21.06" />` in your scene is simply used to specify to the user that this scene was written using SOFA v21.06.  
 
Hence, the message emitted here warns the user that he probably didn't meant that he ran this scene with v21.06 since this version does not exists yet (the largest API version is 20.12.99). 

Now, we could probably talk about this version number thing at the next dev meeting (cc @hugtalbot for the agenda!) . I often see in other projects the master branch having the version number of the next upcoming release. For example, for the next release, it could look like this:
1. `master` branch has version 21.06
2. Close to the release, a `v21.06` branch is created and an alpha or beta binary release is created from this branch
3. `master` branch version is updated to 21.12 (even if 21.06 is not released yet)

Using this, setting `<APIVersion level="21.06" />` in scene files of the master branch would make sense.

Reply:
> Now, we could probably talk about this version number thing at the next dev meeting (cc @hugtalbot for the agenda!) . I often see in other projects the master branch having the version number of the next upcoming release. For example, for the next release, it could look like this:
> 
>     1. `master` branch has version 21.06
> 
>     2. Close to the release, a `v21.06` branch is created and an alpha or beta binary release is created from this branch
> 
>     3. `master` branch version is updated to 21.12 (even if 21.06 is not released yet)
> 
> 
> Using this, setting `<APIVersion level="21.06" />` in scene files of the master branch would make sense.

You are right, the project version was wrong. I used this opportunity to fix it :+1: